functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -258,22 +344,54 @@ Then it will export the results to Export-DbaDiagnosticQuery.
+                            SqlInstance      = $server.DomainInstanceName
+                            Number           = $scriptpart.QueryNr
+                            Name             = $scriptpart.QueryName
+                            Description      = $scriptpart.Description
+                            DatabaseSpecific = $scriptpart.DBSpecific
+                            DatabaseName     = $null
+                            Notes            = "WhatIf - Bypassed Execution"
+                            Result           = $null
                         }
                     }
+
                 }
                 elseif ($scriptpart.DBSpecific -and !$instanceOnly) {
+
                     foreach ($currentdb in $databases) {
-                        $dbname = $currentdb.name
-                        if ($PSCmdlet.ShouldProcess(('{0} ({1})' -f $instance, $currentdb.name), $scriptpart.QueryName)) {
+                        if ($ExportQueries) {
+                            $null = New-Item -Path $OutputPath -ItemType Directory -Force
+                            $FileName = Remove-InvalidFileNameChars ('{0}-{1}-{2}.sql' -f $server.DomainInstanceName, $currentDb, $Scriptpart.QueryName)
+                            $FullName = Join-Path $OutputPath $FileName
+                            Write-Message -Level Verbose -Message  "Creating file: $FullName"
+                            $scriptPart.Text | out-file -FilePath $FullName -encoding UTF8 -force
+                            continue
+                        }
+
+
+                        if ($PSCmdlet.ShouldProcess(('{0} ({1})' -f $instance, $currentDb), $scriptpart.QueryName)) {
+
+                            if (-not $EnableException) {
+                                $Counter++
+                                Write-Progress -Id $ProgressId -ParentId 0 -Activity "Collecting diagnostic query data from $($currentDb) on $instance" -Status ('Processing {0} of {1}' -f $counter, $scriptcount) -CurrentOperation $scriptpart.QueryName -PercentComplete (($Counter / $scriptcount) * 100)
+                            }
 
-                            if (-not $EnableException) { Write-Progress -Id 1 -ParentId 0 -Activity "Collecting diagnostic query data from $dbname on $instance" -Status ('Processing {0} of {1}' -f $counter, $scriptcount) -CurrentOperation $scriptpart.QueryName -PercentComplete (($Counter / $scriptcount) * 100) }
-                            Write-Message -Level Output -Message "Collecting diagnostic query data from $dbname for $($scriptpart.QueryName) on $instance"
+                            Write-Message -Level Verbose -Message "Collecting diagnostic query data from $($currentDb) for $($scriptpart.QueryName) on $instance"
                             try {
-                                $result = $server.Query($scriptpart.Text, $currentdb.Name)
+                                $result = $server.Query($scriptpart.Text, $currentDb)
```

Comment:
The string value of the database name has to be passed into the query method. Revert this back to `$currentDb.Name`